pkg/tests/end_to_end_tests/metric_ingest_bench_test.go

```diff
@@ -74,3 +77,30 @@ func BenchmarkMetricIngest(b *testing.B) {
 		b.StopTimer()
 	})
 }
+
+func BenchmarkNewSeriesIngestion(b *testing.B) {
+	seriesGen, err := testsupport.NewSeriesGenerator(100, 100, 4)
+	require.NoError(b, err)
+
+	batches := seriesGen.GetBatches(2000)
+
+	withDB(b, "bench_e2e_new_series_ingest", func(db *pgxpool.Pool, t testing.TB) {
+		metricsIngestor, err := ingestor.NewPgxIngestorForTests(pgxconn.NewPgxConn(db), &ingestor.Cfg{
+			NumCopiers:              8,
+			InvertedLabelsCacheSize: cache.DefaultConfig.InvertedLabelsCacheSize,
+		})
+		require.NoError(b, err)
+		defer metricsIngestor.Close()
+
+		b.Run("New Series Ingestion", func(b *testing.B) {
+			b.ReportAllocs()
+			for i := 0; i < len(batches); i++ {
+				_, _, _ = metricsIngestor.Ingest(context.Background(), &prompb.WriteRequest{Timeseries: batches[i]})
```

Comment:
Imho it would be useful to also add a benchmark for only metadata ingest, meaning calling `WriteSeries` directly.

Reply:
If we want to benchmark `WriteSeries`, then it should be written in `pgmodel/ingestor/series_writer_test.go` and there, we cannot use the `WithDB`. If we want to do this way, then it will require a lot of changes in the `testhelpers` package to make WithDB accessible to external modules.